sdk/tables/Azure.Data.Tables/api/Azure.Data.Tables.netstandard2.0.cs

```diff
@@ -5,6 +5,7 @@ public partial class TableClient
         protected TableClient() { }
         public virtual Azure.Response<Azure.Data.Tables.Models.TableItem> Create(System.Threading.CancellationToken cancellationToken = default(System.Threading.CancellationToken)) { throw null; }
         public virtual System.Threading.Tasks.Task<Azure.Response<Azure.Data.Tables.Models.TableItem>> CreateAsync(System.Threading.CancellationToken cancellationToken = default(System.Threading.CancellationToken)) { throw null; }
+        public string CreateFilter<T>(System.Linq.Expressions.Expression<System.Func<T, bool>> filter) { throw null; }
```

Comment:
Create Filter sounds a little bit like it's a service operation.

Who would consume this method?

Reply:
The feature crew had a discussion last week about query functionality and the idea was raised to provide a mechanism for developers to easily see the filter query parsed out of their filter expression.

Reply:
Interesting, is it to support mixed query generation scenarios where you generate part of the query and manually format the other part?

Reply:
That could be one, but there were a couple other primary concerns. Firstly, for transparency it would be nice to make it easy for developers to inspect the filter that they generate from the expression. Secondly, this makes it possible to generate and cache a query for later use.

Reply:
> Firstly, for transparency it would be nice to make it easy for developers to inspect the filter that they generate from the expression

Do you think our typical HTTP level logging is not enough in this case?

> Secondly, this makes it possible to generate and cache a query for later use.

Is performance around query generation a know issue that customers hit?

I'm not strictly against exposing this feature just wonder if doing it proactively is worth it and if the client type is the right place for it (as opposed to a less visible static method somewhere).

Reply:
That's fair - Let me try it with an extension method on TableClient